### PR TITLE
Add delete action for folders

### PR DESCRIPTION
**Pull request by GretaD**

### CODE REVIEW

**Comment by jancborchardt** [comment]
I’d say this needs the ability to create a folder as prerequisite as well, no?

**Comment by ChristophWurst** [comment]
We have that ;)

**Comment by GretaD** [comment]
> We have that ;)

and since we're here, can you check if the front end is done right, so i can dig more on the back end site and finish it asap :) same for the account. Of course when you have some time

**Comment by jancborchardt** [comment]
> > We have that ;)
> 
> and since we're here, can you check if the front end is done right, so i can dig more on the back end site and finish it asap :) same for the account. Of course when you have some time

Ah right, noticed it but totally overlooked that! Some feedback on it – also valid for the "Add subfolder" flow:
- [ ] By default, it should only show as normal entry, not directly as input field cause that’s too much noise. Just make it so that on click it converts into the input field and automatically focuses. Then the placeholder can also be "Folder name".
- [ ] A more fitting icon for the "Add folder" entry is `icon-folder`
- [ ] When the folders are collapsed and you add a folder, you don’t see any feedback. So in this case, the folders should automatically expand.
- [ ] Seems like a strange bug possibly connected to the caching: The first folder I created shows the same emails as the inbox, but starting from message number 16 only. But when I click one, it says "Not found: Could not load your message"
- [ ] Guess that’s a new feature, but it’s not possible to rename a folder

**Comment by jancborchardt** [comment]
As for this pull request, the wording of the modal can be improved:

> **Delete folder**
> The folder and all messages in it will be deleted.
> [ Cancel ]   [ Delete folder Testing ]

**Comment by GretaD** [comment]
> See comments :)

Thank you, this PR is not done yet, i pushed it for Christoph to see why it was not working :) but again, thank you for the feedback

**Comment by ChristophWurst** [comment]
Please open tickets for the items in https://github.com/nextcloud/mail/pull/2730#issuecomment-604449943 :)

**Comment by jancborchardt** [comment]
> Please open tickets for the items in [#2730 (comment)](https://github.com/nextcloud/mail/pull/2730#issuecomment-604449943) :)

Done and adjusted the comment: https://github.com/nextcloud/mail/pull/2730#issuecomment-604449943 :)

**Comment by GretaD** [comment]
To be fixed: 

- [ ] add action menu for subfolders

**Comment by GretaD** [comment]
@jancborchardt can you please test this and see if all is good? <3

**Comment by jancborchardt** [comment]
As mentioned in my comment above, the wording of the modal can be improved:

> **Delete folder**
> The folder and all messages in it will be deleted.
> [ Cancel ]   [ Delete folder Testing ]

That is, shorten the main sentence and include the folder name in the delete button for confirmation.

Also, the "Delete" action should _always_ be the last one in the action menu. At the moment, clear cache is below it.

**Comment by GretaD** [comment]
> As mentioned in my comment above, the wording of the modal can be improved:
> 
> > **Delete folder**
> > The folder and all messages in it will be deleted.
> > [ Cancel ]   [ Delete folder Testing ]
> 
> That is, shorten the main sentence and include the folder name in the delete button for confirmation.
> 

Oh i thought i should add this sentence and not this should be the only one. Sorry.

> Also, the "Delete" action should _always_ be the last one in the action menu. At the moment, clear cache is below it.

ok, on it

File: src/store/mutations.js [hunk 1]
```diff
@@ -107,6 +107,14 @@ export default {
 			account.folders.push(id)
 		})
 	},
+	removeFolder(state, {account, folder}) {
+		console.log('something', account, folder)
+		const folderId = normalizedFolderId(account.id, folder.id)
+		Vue.delete(state.folders, folderId)
+		//TODO: Handle subfolders
```
Comment (ChristophWurst) [inline src/store/mutations.js:114]: :shrug:

**Comment by GretaD** [comment]
all is good for final review

**Pull request merged**